File: 1semana/aula3/carro.py
class Carro:
    def __init__(self, modelo, ano):
        self.modelo = modelo
        self.ano = ano
        self.ligado = False

    # Método para ligar o carro
    def ligar(self):
        self.ligado = True

    # Método para desligar o carro
    def desligar(self):
        self.ligado = False

    # Método para retornar os detalhes do carro
    def __str__(self):
        return f"Modelo: {self.modelo}\nAno: {self.ano}\nLigado = {self.ligado}"

# Exemplo de uso
if __name__ == "__main__":
    c = Carro("UNO", 2005)
    print(c)
    c.ligar()
    print(c)
    c.desligar()
    print(c)
    

File: 1semana/aula3/Carro.java


public class Carro {
    private String modelo;
    private int ano;
    private boolean ligado;

    // Construtor
    public Carro(String modelo, int ano) {
        this.modelo = modelo;
        this.ano = ano;
        this.ligado = false;
    }

    // Método para ligar o carro
    public void ligar() {
        ligado = true;
    }

    // Método para desligar o carro
    public void desligar() {
        ligado = false;
    }

    // Método para imprimir os detalhes do carro
    public String toString() {
        return new String("Modelo: " + this.modelo
                        + "\nAno: " + this.ano
                        + "\nLigado = " + this.ligado);
    }
    
    public static void main(String[] args) {
        Carro c = new Carro("UNO", 2005);
        System.out.println(c);
        c.ligar();
        System.out.println(c);
        c.desligar();
        System.out.println(c);
    }
}